K_Scaffold/Template v2/scaffold/tabs/tabs.js

```diff
@@ -2,16 +2,34 @@
  * The default tab navigation function of the K-scaffold. Courtesy of Riernar. It will add `k-active-tab` to the active tab-container and `k-active-button` to the active button. You can either write your own CSS to control display of these, or use the default CSS included in `scaffold/_k.scss`. Note that `k-active-button` has no default CSS as it is assumed that you will want to style the active button to match your system.
  * @param {Object} trigger - The trigger object
  */
-const _kSwitchTab = function ({ trigger }) {
+const kSwitchTab = function ({ trigger, attributes }) {
   const [container, tab] = (
     trigger.name.match(/nav-tabs-(.+)--(.+)/) ||
     []
   ).slice(1);
-  console.log([container, tab]);
   $20(`[data-container-tab="${container}"]`).removeClass('k-active-tab');
   $20(`[data-container-tab="${container}"][data-tab="${tab}"]`).addClass('k-active-tab');
   $20(`[data-container-button="${container}"]`).removeClass('k-active-button');
   $20(`[data-container-button="${container}"][data-button="${tab}"]`).addClass('k-active-button');
+  const tabInputName = `${container.replace(/\-/g,'_')}_tab`;
+  if(persistentTabs.indexOf(tabInputName) > -1){
+    attributes[tabInputName] = trigger.name;
```

Comment:
Wait, you can do that with an AttributeProxy and it calls `setAttrs` on its own ??

Reply:
Yep, that's how I set up the attribute proxy and the default listener architecture. Provides a more streamlined interface for getting / setting attributes. The default listener does a set using all the accrued updates once everything has run.

Reply:
That's sooo nice !
And when it's in the "a modification is scheduled, but not yet applied" state, does accessing the attribute gives you the upcoming value or the old one ?

Reply:
The update.

The proxy essentially has two properties that store attribute information (which I horribly flubbed the naming of, but live and learn):

- `attributes`: The original attribute data as it was when you ran `k.getAttrs` or `k.getAllAttrs`.
- `updates`: Any modifications that have been done to the attribute data since then. Currently it only stores the last update that was made, not the entire modification history for a given run.

Any time you do something like `const myStrength = attributes.strength;`, it gives you the value in the `updates` property, or the value in `attributes` if no update has been given yet. You can also always access/modify the original state by specifying `attributes.attributes.strength`.

Reply:
You could also specifically access the updates version by doing `attributes.updates.strength`. Not real sure there's a use case for that. Maybe if you had some real complicated dynamic setup that needed to do something if it hadn't already been done? Regardless, it's a fundamental part of the architecture so it's not going anywhere.

Reply:
I should note that if you're doing your own custom listener or something, it won't set it to the sheet automatically, but if you're using the default `accessSheet` listener, it will. Essentially the function cascade started by `accessSheet` does:

```js
k.getAllAttrs({
  callback:(attributes,sections,casc)=>{
    // Invoke functions for initial change if this is the trigger corresponding to the triggering event
    // Invoke triggered functions
    // Invoke calculation functions
    attributes.set();
  }
});
```
The actual setup is much more recursive than that, but that's the process distilled down to it's essential steps.